### ISSUE TITLE
Feature Request removeUsersFromGroups

### ISSUE THREAD

**Issue by pgrm** [alanning/meteor-roles#155]
Currently I don't see any way to remove a user from a group to which the user has been assigned before. (of course, except actually removing the property from the user object, but an API for it would be nice)

**Comment by alanning** [comment]
This gets into a bit of philosophy on what this package should cover.  Currently "membership" in a group is not explicitly tracked by this package.  Rather, only the permissions that users have within a group are tracked.

So for example, if we execute the following...
```
Roles.addUsersToRoles(joesUserId, 'manage-team', 'manchester-united.com')
Roles.removeUsersFromRoles(joesUserId, 'manage-team', 'manchester-united.com')
```
... is Joe still a member of Manchester United or not?

Roles currently doesn't make any decisions on membership.  In my own app, we have a separate "channels" field on the user object which tracks membership and we use roles to track permissions for the channels that the user is a member of.  This is also important for performance reasons because the channels field may be indexed.

Roles v2.0 may also track membership as it is a commonly needed thing to do.  Since it will allow indexing on group, we may just encourage people to use a specific role like "member" to explicitly track the membership.

What are your thoughts on this?

**Comment by pgrm** [comment]
That's what I'm doing. My lowest role is guest so that's how I track membership.

The reason behind removing groups from the user object is purely a concern, to not have unbounded arrays in a document. While in my application a user will be at any given time probably only part of few groups, throughout it's lifetime the user could join (temporarily) hundreds or thousands of groups and again leave them, which would leave me with an object full of empty properties.

Maybe my usage of groups is wrong in this case or it's just premature optimization (I can't really prove that in reality the user would be joining hundreds of groups)

But I'm definitely looking forward to v2

**Comment by alanning** [comment]
Ah, I see.  One way to approach it would be to make a nightly script that searches through and removes empty groups.  Another way would be to monkey-patch the `removeUsersFromRoles` and do a check after it completes; if the user has no more roles in that group, perform an `$unset` on that group field.

Not ideal, but that should at least help mitigate the risk of all those empty fields.  :-)

**Comment by mitar** [comment]
In v2.0 (currently in development) this is not an issue anymore because the database schema is different. When you remove the last role with a given group from the user, there is no information about that group being linked with that user anymore in its document.

I am closing this because it is resolved in v2.0.
